plugin/src/test/java/dev/regadas/trino/pubsub/listener/PubSubEventListenerTest.java

```diff
@@ -155,7 +157,8 @@ void testQueryCreatedPublishCorrespondingMessage() {
 
         eventListener.queryCreated(TestData.FULL_QUERY_CREATED_EVENT);
 
-        assertThat(publisher.lastPublishedMessage, instanceOf(Schema.QueryCreatedEvent.class));
+        assertThat(publisher.lastPublishedMessage, instanceOf(Schema.QueryEvent.class));
+        assertThat(parseQueryEvent(publisher.lastPublishedMessage).hasQueryCreated(), is(true));
```

Comment:
:thinking: how does this works if line 153 sets a json encoding?

Reply:
[Here](https://github.com/regadas/trino-pubsub-event-listener/blob/main/plugin/src/test/java/dev/regadas/trino/pubsub/listener/PubSubEventListenerTest.java#L243) `lastPublishedMessage` is captured before `publish` is called. It's an unserialised Message object that can be casted to/back QueryEvent.